### ISSUE TITLE
Unable to replicate paper results

### ISSUE THREAD

**Issue by purvanshi** [A2Zadeh/TensorFusionNetwork#5]
Hey,
I am unable to replicate the results (get a binary test accuracy of 77.1) I made the following changes of hyperparameters in [this](https://github.com/Justin1904/TensorFusionNetworks) code -

1) Changed the output of the fusion layer to 128 instead of 32
2) dropout of 0.15 as stated in the paper
3) added l2 norm regulariser by adding a weight decaying factor of 0.01
4) Changed the number of output nodes of the text modality to 64 instead of 32
I run the model with a batch_size of 128, 1000 epochs and a patience of 100
Am I missing something here?

**Comment by A2Zadeh** [comment]
Hello @purvanshi. So is this on the new train,test fold?

**Comment by A2Zadeh** [comment]
@purvanshi I was going through the comments on github. Were you able to replicate the results? If not please let me know. I would be interested to know why that was the case.

**Comment by purvanshi** [comment]
Hello @A2Zadeh I was not able to replicate the results. I tried several hyperparameters.

**Comment by A2Zadeh** [comment]
@purvanshi oh sorry this fell through the cracks. I am happy to help recreate the results. Can you answer a few questions for me?

1. Are you planning on recreating the cross-validation version of the original paper? Currently we do a holdout. I can help recreate either but if you are pushing for a paper maybe the second option is what you are looking for. Results are available here: https://arxiv.org/pdf/1802.00923.pdf

2. Are you using the feature selection?

3. What are the set of hypers you are trying?

**Comment by purvanshi** [comment]
@A2Zadeh thanks for your reply
1) I am building upon https://arxiv.org/abs/1707.07250 this paper. 
2) What exactly do you mean by feature selection here?
3) The hyperparameters -
   dropout of 0.15 as stated in the paper
   added l1 norm regulariser by adding a weight decaying factor of 0.01
  32 as the output for text modality
  Batchsize of 64

These are the hyperparameters I am getting the best accuracy on.

**Comment by A2Zadeh** [comment]
@purvanshi I see. So basically original Tensor Fusion used cross-validation. Currently methods are using a holdout data. It's up to you which one you prefer to do, but majority of recent work is using holdout (such as the paper I sent), in case you want to easily compare to them. 

Yes, there is a feature selection. Have a look at the data here: https://github.com/pliang279/MFN/issues/4
The data is also available on the same GitHub repo. 

I would suggest a hyper parameter search space of at least 100 different combinations. Keep me in the loop :)

**Comment by A2Zadeh** [comment]
@purvanshi any updates? Still not able to recreate the performance?

**Comment by A2Zadeh** [comment]
@purvanshi Just wanted to double check if you were able to recreate the results, or if you had any questions about how to recreate them. Please let me know.
